src/hidimstat/base_perturbation.py

```diff
@@ -139,27 +152,69 @@ def importance(self, X, y):
         """
         self._check_fit(X)
 
-        out_dict = dict()
-
         y_pred = getattr(self.estimator, self.method)(X)
-        loss_reference = self.loss(y, y_pred)
-        out_dict["loss_reference"] = loss_reference
+        self.loss_reference_ = self.loss(y, y_pred)
 
         y_pred = self.predict(X)
-        out_dict["loss"] = dict()
+        self.loss_ = dict()
         for j, y_pred_j in enumerate(y_pred):
             list_loss = []
             for y_pred_perm in y_pred_j:
                 list_loss.append(self.loss(y, y_pred_perm))
-            out_dict["loss"][j] = np.array(list_loss)
+            self.loss_[j] = np.array(list_loss)
 
-        out_dict["importance"] = np.array(
+        self.importances_ = np.array(
             [
-                np.mean(out_dict["loss"][j]) - loss_reference
-                for j in range(self.n_groups)
+                np.mean(self.loss_[j]) - self.loss_reference_
+                for j in range(self._n_groups)
             ]
         )
-        return out_dict
+        self.pvalues_ = ttest_1samp(
+            self.importances_, 0.0, axis=0, alternative="greater"
+        ).pvalue
+        return self.importances_
+
+    def fit_importance(
```

Comment:
I find it disturbing that `fit_importance` has a behavior that is quite different from simply calling fit, then importance. 
 - Could we add a check to the `.fit()` method to ensure that the estimator is fitted, and if not, fit it. 
 - Could we allow for passing a list of fitted estimators matching the number of splits? That could typically be relevant for users willing to pass DL models, trained before, through skorch for instance. 
 - If the models are not fitted, can we store them? in estimators_ for instance? It is useful to check the predictive performance in addition to the importance.

Reply:
I did the modification, tell me it's ok.